common/display/src/main/scala/uk/ac/wellcome/display/json/DisplayJsonSerializers.scala

```diff
@@ -0,0 +1,22 @@
+package uk.ac.wellcome.display.json
+
+import io.circe.Encoder
+import uk.ac.wellcome.display.models.SortingOrder.{Ascending, Descending}
+import uk.ac.wellcome.display.models.{
+  ProductionDateSortRequest,
+  SortRequest,
+  SortingOrder
+}
+
+trait DisplayJsonSerializers {
```

Comment:
Maybe `DisplayParameterSerializers`, or something like that, would be a more appropriate name? The current name doesn't really differentiate these from all the serializers defined elsewhere, and communicate that they for the purpose of serializing input query parameters.

Reply:
What do you think about this method over having the `SourceData => InternalModel => DisplayModel` and rather replacing `DisplayModel` with serialisers?

[There's some examples and discussion here](https://github.com/wellcomecollection/stacks-service/pull/49#issuecomment-558740047).

TL;DR - it feels like what we've invented as `DisplayModels` is actually just serialisers.
- They should live on the edge of the application
- we should not really use them internally
- we can do it consistently without having to remember referenced models in `DisplayModels`
- we can create different serialiser contexts (for now we just have `ApiDisplaySerialisers`or something like that)

The reason this was called this was to be more generic and see how we felt about it - I can split them up similar to the `DisplayModels` though.

Reply:
For me the display models provide a way to understand what will be serialised. Skipping that steps makes it less easy to see in the code what the serialised version will look like. It also provides a serialisation free model of what we are making available: for example if you wanted to serialise the same model to XML & JSON an intermediate display abstraction would be useful.

I don't think we need to apply the "display model" approach in all cases, especially where there are small simple models being serialised.  **To further the experiment** trying a complex model would be informative as to whether removing display models extensively is beneficial.

Reply:
:+1: I agree the display models introduces an extra layer of complexity which could maybe be avoided.

I need to think more about this question in general, and probably worth us all having a conversation about it at some point. One thing that springs to mind is that we will likely still need lots of these models defined somewhere, even if they are pushed further to the edge of the application, due to the fact that using case classes and circe derivation is a really useful and declaritive method to define serializers / deserializers.

Reply:
I think I prefer having the DisplayModel intermediate steps. I agree that they add complexity but, as @kenoir is saying, they make it much easier to understand what gets serialised. Also, they allow for (relatively) easy swagger documentation. 
I also disagree that they are just serialisation. The serialisation is done by circe, and we could (relatively) easily swap the circe for another json library or add another type serialisation with minimal changes to the display models. I think the DisplayModels are models and if we skip this step, we _might_ end up with less lines of code (although not convinced about that because of swagger and circe etc) but we collapse two purposes - serialisation and model representation - making our life a bit harder.

Reply:
@alicefuzier I'll swap it out.